src/trio/_core/_tests/test_asyncgen.py

```diff
@@ -68,20 +71,20 @@ async def async_main() -> None:
         # No problems saving the geniter when using either of these patterns
         aiter_ = example("exhausted 3")
         try:
-            saved.append(aiter_)
+            saved.append(cast("AGT", aiter_))
```

Comment:
IMO an `isinstance` check later is better than a `cast` here.

... I assume `example` can't return `AsyncGeneratorType[int, None]`?

Reply:
> `example` can't return `AsyncGeneratorType[int, None]`?

Nope! You get a different mypy error:
`trio/_core/_tests/test_asyncgen.py:22: error: The return type of an async generator function should be "AsyncGenerator" or one of its supertypes  [misc]`

> an `isinstance` check later

I tried this and got an error, but upon trying again, I no longer get an error. Not sure what happened the first time. Thanks!